compose/service/record.go

```diff
@@ -354,71 +355,225 @@ func (svc record) Export(filter types.RecordFilter, enc Encoder) error {
 	return set.Walk(enc.Record)
 }
 
-func (svc record) Create(new *types.Record) (rec *types.Record, err error) {
-	var invokerID = auth.GetIdentityFromContext(svc.ctx).Identity()
-
+// Operate handles provided set of bulk record operations.
+// It's able to create or update records in a single transaction.
+//
+// Feature used mainly by Record Lines, but should be used when ever we wish to perform
+// multiple operations over records.
+func (svc record) Operate(pl ...*types.BulkRecordOperation) (rec types.RecordSet, err error) {
 	return rec, svc.db.Transaction(func() (err error) {
-		var (
-			ns *types.Namespace
-			m  *types.Module
-		)
+		for _, p := range pl {
+			r := p.Record
 
-		ns, m, _, err = svc.loadCombo(new.NamespaceID, new.ModuleID, 0)
-		if err != nil {
-			return
-		}
-
-		if !svc.ac.CanCreateRecord(svc.ctx, m) {
-			return ErrNoCreatePermissions.withStack()
-		}
+			// Handle any pre processing, such as defining parent recordID.
+			// This functions are defined by the system, so there is no need for validation
+			//
+			// @todo This should be moved into svc.create(...) and svc.update(...) functions,
```

Comment:
why it should be moved? pre-process is (as I understand it) part of bulk op)

Reply:
This should become a post-process, that happens right before the values are saved to the DB.
I temporarily changed it to pre-process, because I can't think of an elegant way to do this at the moment.